src/ips/peer.rs

```diff
@@ -33,7 +33,7 @@ impl Peer {
         };
 
         for peer in &node.connections {
-            if *peer >= nodes.len() {
+            if *peer >= nodes.len() || nodes[*peer].addr == node.addr {
```

Comment:
Yes, there are some paths that run peerlist generation right after re-constructing graph. As spectre does not allow to add single nodes I'm adding nodes connected to themselves. Then, if we'd like to generate peerlist it would contain such connections so this place get rid of them.